tunnistamo/tests/test_oidc_provider_backchannel_logout.py

```diff
@@ -0,0 +1,294 @@
+import logging
+from timeit import default_timer
+
+import jwt
+import pytest
+from django.test import Client as DjangoTestClient
+from django.test.client import RequestFactory
+from django.urls import reverse
+from httpretty import httprettified, httpretty
+from oidc_provider.lib.utils.token import create_token
+
+from oidc_apis.backchannel_logout import send_backchannel_logout_to_apis_in_token_scope
+from oidc_apis.factories import ApiDomainFactory, ApiFactory, ApiScopeFactory
+from oidc_apis.models import ApiScope
+from tunnistamo.oidc import create_logout_token
+from tunnistamo.tests.conftest import create_oidc_clients_and_api, get_tokens, reload_social_django_utils
+from users.models import TunnistamoSession
+
+
+def fix_httpretty_latest_requests_list(latest_requests):
+    """Deduplicate POST requests in HTTPretty's latest requests list
+
+    HTTPretty has introduced a bug where POST requests are doubled in the request
+    history. See Issue in their repository: https://github.com/gabrielfalcao/HTTPretty/issues/425"""
+    from httpretty import version as httpretty_version
+
+    if httpretty_version < '1.1' or httpretty_version > '1.1.4':
```

Comment:
1.1.4 is currently the latest release of httpretty. It isn't certain that this issue will be fixed in the next release. Should here be some code that whoever updates httpretty dependency next, has to check what's the status of this issue? Something like `if httpretty_version > '1.1.4': assert False` (with an explanatory description what to do).

Reply:
Good point. I'll add it.